web/src/utils/formatting.ts

```diff
@@ -19,18 +18,37 @@ export const formatPower = function (
   return power;
 };
 
-export const formatCo2 = function (d: number, numberDigits: number = DEFAULT_NUM_DIGITS) {
-  let value = d;
+export const formatCo2 = function (gramPerHour: number, valueToMatch?: number) {
+  if (gramPerHour == undefined || Number.isNaN(gramPerHour)) {
+    return gramPerHour;
+  }
+
   // Assume gCO₂ / h input
+  let value = gramPerHour;
   value /= 60; // Convert to gCO₂ / min
   value /= 1e6; // Convert to tCO₂ / min
-  if (d == undefined || Number.isNaN(d)) {
-    return d;
+
+  // Ensure both numbers are at the same scale
+  const checkAgainst = valueToMatch ? valueToMatch / 1e6 : value;
+
+  // grams and kilograms
+  if (Math.round(checkAgainst) < 1) {
+    return `${d3.format(`,.0~s`)(value * 1e6)}g`;
+  }
+
+  // tons
+  if (Math.round(checkAgainst) < 1e5) {
+    const decimals = value < 1 ? 1 : 0;
+    return `${d3.format(`,.${decimals}~f`)(value)}t`;
+  }
+
+  // Hundred thousands of tons
+  if (Math.round(checkAgainst) < 1e6) {
+    return `${d3.format(`,.1~f`)(value / 1e6)}Mt`;
   }
 
-  return value >= 1
-    ? `${d3.format(`.${numberDigits}s`)(value)}t ${translate('ofCO2eqPerMinute')}` // a ton or more
-    : `${d3.format(`.${numberDigits}s`)(value * 1e6)}g ${translate('ofCO2eqPerMinute')}`;
+  // megatons or above
+  return `${d3.format(`,.2~s`)(value)}t`;
```

Comment:
We can fix the missing space, but it requires using regex to split the result and then put a space in before joining it together again. This is because d3-format returns `4M` when using the SI-prefix